hibernate-core/src/main/java/org/hibernate/loader/hql/QueryLoader.java

```diff
@@ -450,9 +449,24 @@ protected Object getResultColumnOrRow(
 		if ( hasScalars ) {
 			String[][] scalarColumns = scalarColumnNames;
 			int queryCols = queryReturnTypes.length;
+			int managedTypeIndex = 0;
+			SessionFactoryImplementor factory = session.getFactory();
 			resultRow = new Object[queryCols];
 			for ( int i = 0; i < queryCols; i++ ) {
-				resultRow[i] = queryReturnTypes[i].nullSafeGet( rs, scalarColumns[i], session, null );
+				// Types for inverse relationships have a zero column span in which case we have to get the
+				// result from the row array.
+				Type queryReturnType = queryReturnTypes[i];
+				if ( queryReturnType.getColumnSpan( factory ) == 0 ) {
```

Comment:
Sure that you don't have to account for `includeInSelect` here? Perhaps because this is a scalar query? What happens when you ie. add a `FETCH JOIN` to the query?